loki/transformations/parametrise.py

```diff
@@ -87,10 +87,60 @@
 from loki.ir import nodes as ir, Transformer, FindNodes
 from loki.tools.util import as_tuple, CaseInsensitiveDict
 
+from loki.transformations.utilities import single_variable_declaration
 from loki.transformations.inline import inline_constant_parameters
 
 
-__all__ = ['ParametriseTransformation']
+__all__ = ['ParametriseTransformation', 'parametrise_single_val_scalars']
+
+
+def parametrise_single_val_scalars(routine):
+    """
+    Parametrise those scalars which are only written to once and assigned a constant value.
+
+    This is not really sophisticated and will eventually be superseded by
+    a constant propagation transformation.
+    """
+    def is_constant_rhs(expr):
+        # expr is a literal e.g., a IntLiteral
+        if issubclass(type(expr), sym._Literal):
+            return True
+        # expr is a Product/Sum and all children are literals
+        if isinstance(expr, (sym.Product, sym.Sum)):
+            return all(issubclass(type(_expr), sym._Literal) for _expr in expr.children)
+        return False
+
+    assignments = FindNodes(ir.Assignment).visit(routine.body)
+    # local variables
+    variables = [var for var in routine.variables if var not in routine.arguments]
+    assignments_dic = {}
+    for assignment in assignments:
+        if assignment.lhs in variables:
+            assignments_dic.setdefault(assignment.lhs, []).append(assignment)
```

Comment:
I'm not sure but you might be running into case-sensitivity issues here. Could be better to build `assignments_dic` as a `CaseInsensitiveDict` with `assignment.lhs.name` as keys